### PR TITLE
Create team repository

### PR DESCRIPTION
**Pull request by joao-gouveia**
This PR adds the team repository.

### CODE REVIEW

File: internal/repositories/postgresql/team.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	INSERT INTO teams (id, name)
+	VALUES($1,$2)
+`
+
+// TaskStorage postgreSQL implementation of a task storage
+type TeamStorage struct {
+	db *sql.DB
+}
+
+// NewTaskStorage returns and instance on the postgreSQL task storage
```
Comment (romeufcrosa) [inline internal/repositories/postgresql/team.go:31]: Comment name and function name should match

File: internal/repositories/postgresql/team.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	FROM teams
+	WHERE id = $1
+`
+
+const InsertTeamQuery = `
+	INSERT INTO teams (id, name)
+	VALUES($1,$2)
+`
+
+// TaskStorage postgreSQL implementation of a task storage
```
Comment (romeufcrosa) [inline internal/repositories/postgresql/team.go:26]: Comment name mismatch here also

File: internal/domain/team.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package domain
+
+import "github.com/google/uuid"
+
+type Team struct {
```
Comment (romeufcrosa) [inline internal/domain/team.go:5]: Please add comment to exported method

File: internal/repositories/postgresql/team.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		return domain.Team{}, err
+	}
+
+	return team, nil
+}
+
+// Create creates a new task
+func (t TeamStorage) Create(team domain.Team) error {
+	_, err := t.db.Exec(InsertTeamQuery,
+		team.ID.String(),
```
Comment (samuelfvlcastro) [inline internal/repositories/postgresql/team.go:88]: I don't think you need to do the `String()`

File: internal/repositories/postgresql/team.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
+package postgresql
+
+import (
+	"database/sql"
+
+	"github.com/Taskdesk/scheduler/internal/domain"
+	"github.com/google/uuid"
+)
+
+const ListTeamQuery = `
```
Comment (samuelfvlcastro) [inline internal/repositories/postgresql/team.go:10]: make it private pls

File: internal/repositories/postgresql/team.go [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	SELECT id, name
+	FROM teams
+`
+
+const SelectTeamQuery = `
```
Comment (samuelfvlcastro) [inline internal/repositories/postgresql/team.go:15]: make it private pls

File: internal/repositories/postgresql/team.go [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+	SELECT id, name
+	FROM teams
+	WHERE id = $1
+`
+
+const InsertTeamQuery = `
```
Comment (samuelfvlcastro) [inline internal/repositories/postgresql/team.go:21]: make it private pls

File: internal/repositories/postgresql/team.go [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+			return []domain.Team{}, err
+		}
+
+		teamList = append(teamList, team)
+	}
+
+	return teamList, nil
+}
+
+// FindByID finds a task by uuid
```
Comment (samuelfvlcastro) [inline internal/repositories/postgresql/team.go]: teams not tasks

File: internal/repositories/postgresql/team.go [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	if err != nil {
+		return domain.Team{}, err
+	}
+
+	return team, nil
+}
+
+// Create creates a new task
```
Comment (samuelfvlcastro) [inline internal/repositories/postgresql/team.go]: team no task

File: internal/repositories/postgresql/team.go [hunk 10]
```diff
@@ -0,0 +1,97 @@
+package postgresql
+
+import (
+	"database/sql"
+
+	"github.com/Taskdesk/scheduler/internal/domain"
+	"github.com/google/uuid"
```
Comment (samuelfvlcastro) [inline internal/repositories/postgresql/team.go:7]: you're missing the postgres blank import

**Pull request merged**